### PR TITLE
Add missing exceptions documentation related to storage_backend

### PR DESCRIPTION
**Pull request by MVrachev**
### Description of the changes being introduced by the pull request:

The missing exceptions documentation that was added is related to the
code for the "storage_backend" and its API calls.

Signed-off-by: Martin Vrachev <mvrachev@vmware.com>

### CODE REVIEW

File: securesystemslib/hash.py [hunk 1]
```diff
@@ -343,6 +343,8 @@ def digest_filename(filename, algorithm=DEFAULT_HASH_ALGORITHM,
     securesystemslib.exceptions.UnsupportedLibraryError, if the given
     'hash_library' is unsupported.
 
+    securesystemslib.exceptions.StorageError, if the file cannot be open.
```
Comment (adityasaky) [inline securesystemslib/hash.py:346]:
```suggestion
    securesystemslib.exceptions.StorageError, if the file cannot be opened.
```

File: securesystemslib/util.py [hunk 2]
```diff
@@ -109,6 +109,9 @@ def get_file_hashes(
 
     securesystemslib.exceptions.Error: If 'filepath' does not exist.
 
+    securesystemslib.exceptions.StorageError: The file at "filepath" cannot be
+    open or found.
```
Comment (adityasaky) [inline securesystemslib/util.py:113]:
```suggestion
    opened or found.
```

**Review by adityasaky — APPROVED** [review]
Nits, but LGTM!

**Comment by MVrachev** [comment]
> Nits, but LGTM!

I updated the pr addressing your comments and updated the exceptions in the test.
Plus added a small use case that wasn't tested.

**Comment by MVrachev** [comment]
PS: The error happening on build https://github.com/secure-systems-lab/securesystemslib/runs/4884928637?check_suite_focus=true is not related to my changes.

**Comment by adityasaky** [comment]
Re-run seems to have fixed it :)

**Pull request merged**